include/ck_tile/ops/gemm/kernel/batched_gemm_kernel.hpp

```diff
@@ -0,0 +1,220 @@
+                                                                   index_t N,
+                                                                   index_t K,
+                                                                   index_t stride_A,
+                                                                   index_t stride_B,
+                                                                   index_t stride_C,
+                                                                   index_t batch_stride_A,
+                                                                   index_t batch_stride_B,
+                                                                   index_t batch_stride_C,
+                                                                   index_t batch_count)
+    {
+        return BatchedGemmCommonKargs{a_ptr,
+                                      b_ptr,
+                                      c_ptr,
+                                      k_batch,
+                                      M,
+                                      N,
+                                      K,
+                                      stride_A,
+                                      stride_B,
+                                      stride_C,
+                                      batch_stride_A,
+                                      batch_stride_B,
+                                      batch_stride_C,
+                                      batch_count};
+    }
+
+    CK_TILE_HOST_DEVICE static constexpr index_t GetSmemSize()
+    {
+        return max(GemmPipeline::GetSmemSize(), EpiloguePipeline::GetSmemSize());
+    }
+
+    CK_TILE_DEVICE void operator()(BatchedGemmCommonKargs kargs) const
+    {
+        const auto [i_m, i_n] = TilePartitioner{}();
+        const auto i_batch    = blockIdx.z;
+        //  options
+        const ADataType* a_start = static_cast<const ADataType*>(kargs.a_ptr) +
+                                   __builtin_amdgcn_readfirstlane(i_batch * kargs.batch_stride_A);
+        const BDataType* b_start = static_cast<const BDataType*>(kargs.b_ptr) +
+                                   __builtin_amdgcn_readfirstlane(i_batch * kargs.batch_stride_B);
+        // Convert pointers to tensor views
+        auto a_tensor_view = [&]() {
+            if constexpr(std::is_same_v<ALayout, tensor_layout::gemm::RowMajor>)
+            {
+                return make_naive_tensor_view<address_space_enum::global>(
+                    a_start,
+                    make_tuple(kargs.M, kargs.K),
+                    make_tuple(kargs.stride_A, 1),
+                    number<GemmPipeline::VectorSizeA>{},
+                    number<1>{});
+            }
+            else
+            {
+                return make_naive_tensor_view<address_space_enum::global>(
+                    a_start,
+                    make_tuple(kargs.M, kargs.K),
+                    make_tuple(1, kargs.stride_A),
+                    number<GemmPipeline::VectorSizeA>{},
```

Comment:
This is a bug, you can't use vector read on a not contiguous dimension. Take a look in here: https://github.com/ROCm/composable_kernel/blob/develop/include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp#L77-L95